### PR TITLE
Fix conditional statements and asserts

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/helper/tasks/CreatePdfFromServletThread.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public void run() {
                 logger.error("URI " + this.targetFolder + this.getProcess().getTitle() + ".PDF-ERROR.log is malformed",
                     e1);
             } catch (IOException e1) {
-                logger.error("Ressource " + uri + " could not be created", e);
+                logger.error("Resource could not be created", e);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/helper/tasks/CreatePdfFromServletThread.java]: Why did you remove the name / content / value of not created URI? This could help on search for error reason.

File: Kitodo/src/main/java/org/goobi/production/model/bibliography/course/Block.java [hunk 2]
```diff
@@ -263,7 +264,7 @@ public boolean isEmpty() {
      * @return whether the given string is equals to the assigned variant
      */
     public boolean isIdentifiedBy(String variant) {
-        return variant == null && this.variant == null || this.variant.equals(variant);
+        return variant == null && this.variant == null || Objects.nonNull(this.variant) && this.variant.equals(variant);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/model/bibliography/course/Block.java]: Should comparing to null not replaced by `Objects.isNull` or `Objects.nonNull`?

File: Kitodo/src/main/java/de/sub/goobi/helper/tasks/CreatePdfFromServletThread.java (hunk 1 above)
Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/helper/tasks/CreatePdfFromServletThread.java]: uri is null here

Comment (henning-gerhardt) [reply Kitodo/src/main/java/de/sub/goobi/helper/tasks/CreatePdfFromServletThread.java]: I see it now and removing is ok in this case.
